hazelcast/src/main/java/com/hazelcast/spi/impl/NodeEngineImpl.java

```diff
@@ -240,9 +240,9 @@ public void start() {
 
         metricsRegistry.collectMetrics(operationService, proxyService, eventService, operationParker);
 
+        operationService.start();
```

Comment:
Can you elaborate why this change in service startup order was necessary?

Reply:
Problem is, `DistributedScheduledExecutorService` registers partition lost listener in `init()` phase. When `invokeOnStableClusterSerial()` is called, it expects OperationService to be up. That's why I moved `operationService.start()` to above `serviceManager.start()`.

Reply:
I think I can solve this issue without changing service start order, by returning immediately from `invokeOnStableClusterSerial()` if node is not joined the cluster yet.

Reply:
Sounds right